source/common/upstream/health_checker_impl.cc

```diff
@@ -270,20 +270,43 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onResponseComplete() {
 
   // It is possible for this session to have been deferred destroyed inline in handleFailure()
   // above so make sure we still have a connection that we might need to close.
```

Comment:
Can you move this comment down into the function about the client null check?

Reply:
Yup, done.